lib/java/com/google/android/material/button/MaterialButton.java

```diff
@@ -333,7 +333,7 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
     }
 
     Paint textPaint = getPaint();
-    int textWidth = (int) textPaint.measureText(getText().toString());
+    int textWidth = getLayout().getWidth();
```

Comment:
int textWidth = Math.min((int) textPaint.measureText(getText().toString()), getLayout().getWidth());

Reply:
Good point, updated.

(Although I still wonder if there is a better way to do this w/o using `Paint#measureText()`...)